test/flink_test.go

```diff
@@ -120,3 +142,139 @@ func (s *CLITestSuite) TestFlink_Autocomplete() {
 		s.runIntegrationTest(test)
 	}
 }
+
+func (s *CLITestSuite) TestFlinkShell() {
+	// HACK: empty test to log in, before running the actual integration test
+	test := CLITest{
+		login: "cloud",
+	}
+	s.runIntegrationTest(test)
+
+	// create a file for go-prompt to use as the input stream
+	file, err := os.Create("input.txt")
+	require.NoError(s.T(), err, "error creating file")
+	defer cleanup(file)
+
+	// set the go-prompt file input env var, so go-prompt uses this file as the input stream
+	err = os.Setenv(prompt.EnvVarInputFile, file.Name())
+	require.NoError(s.T(), err, "failed to set go-prompt input file env var")
+
+	// fake the timezone, to ensure CI and local run with the same default timezone
+	err = os.Setenv(timezoneEnvVar, "Europe/London")
+	require.NoError(s.T(), err, "failed to set timezone env var")
+
+	// start flink shell
+	dir, err := os.Getwd()
+	require.NoError(s.T(), err)
+	cmd := exec.Command(filepath.Join(dir, testBin), "flink", "shell", "--compute-pool", "lfcp-123456")
+
+	var outputBuffer bytes.Buffer
+	cmd.Stdout = &outputBuffer
+	cmd.Stderr = &outputBuffer
+
+	err = cmd.Start()
+	require.NoError(s.T(), err, outputBuffer.String())
+
+	// wait for flink shell to be ready
+	time.Sleep(5 * time.Second)
```

Comment:
Is there a better way to verify this?

Reply:
hmm maybe.. Perhaps we could wait for the `Welcome` text? 🤔